ui/js/controllers/bugfiler.js

```diff
@@ -163,41 +184,92 @@ treeherder.controller('BugFilerCtrl', [
+                $http.get(hgBaseUrl + "mozilla-central/json-mozbuildinfo?p=" + failurePath).then(function(firstRequest) {
+                    if (firstRequest.data.aggregate && firstRequest.data.aggregate.recommended_bug_component) {
+                        var suggested = firstRequest.data.aggregate.recommended_bug_component;
+                        addProduct(suggested[0] + " :: " + suggested[1]);
                     }
 
                     $scope.searching = false;
 
-                    if ($scope.suggestedProducts.length === 0) {
-                        var jg = selectedJob.job_group_name.toLowerCase();
-                        // Some job types are special, lets explicitly handle them.
-                        if (jg.includes("web platform")) {
-                            $scope.suggestedProducts.push("Testing :: web-platform-tests");
-                        }
-                        if (jg.includes("talos")) {
-                            $scope.suggestedProducts.push("Testing :: Talos");
-                        }
-                        if (jg.includes("mochitest") && failurePath.includes("webextensions/")) {
-                            $scope.suggestedProducts.push("Toolkit :: WebExtensions: General");
-                        }
-                        if (jg.includes("mochitest") && failurePath.includes("webrtc/")) {
-                            $scope.suggestedProducts.push("Core :: WebRTC");
-                        }
+                    // Make an attempt to find the file path via a dxr file search
+                    if ($scope.suggestedProducts.length === 0 && $uibModalInstance.possibleFilename.length > 4) {
+                        $scope.searching = "DXR & Mercurial";
+                        var dxrlink = dxrBaseUrl + "mozilla-central/search?q=file:" + $uibModalInstance.possibleFilename + "&redirect=false&limit=5";
+                        // Bug 1358328 - We need to override headers here until DXR returns JSON with the default Accept header
+                        $http.get(dxrlink, {"headers": {
+                            "Accept": "application/json"
+                        }}).then(function(secondRequest) {
+                            var results = secondRequest.data.results;
+                            var resultsCount = results.length;
+                            // If the search returns too many results, this probably isn't a good search term, so bail
+                            if (resultsCount === 0) {
+                                $scope.searching = false;
+                                injectProducts(failurePath);
+                            }
+                            for (var i = 0; i < results.length; i++) {
+                                $scope.searching = "DXR & Mercurial";
+                                $http.get(hgBaseUrl + "mozilla-central/json-mozbuildinfo?p=" + results[i].path).then(function(thirdRequest) {
```

Comment:
better to line wrap at the ``.then`` and use templating and arrow function syntax:

    $http.get(`${hgBaseUrl}mozilla-central/json-mozbuildinfo?p=${results[i].path}`)
        .then(thirdRequest => {